src/types.ts

```diff
@@ -234,8 +236,16 @@ export type StateUpdateCallback<Data = any, Error = any> = (
   isValidating?: boolean
 ) => void
 
-export interface Cache<Data = any> {
+export interface ICache<Data = any> {
   get(key: Key): Data | null | undefined
   set(key: Key, value: Data): void
   delete(key: Key): void
 }
+
+export type DefaultProvider = Map<string | null, any>
```

Comment:
Why `string | null` instead of just `string`?

Reply:
because `swr/infinite` has some case that cache key is `string | null`

like

```ts
  const resolvePageSize = useCallback((): number => {
    const cachedPageSize = cache.get(pageSizeCacheKey)
    return isUndefined(cachedPageSize) ? initialSize : cachedPageSize
```

Reply:
That feels like a bug (facepalm)